src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { UpdateTodoArgs } from '../../types/UpdateTodoArgs';
+
+interface Props {
+  todo: Todo;
+  deleteTodo: (todoId: number) => void;
+  toggleTodoStatus:(
+    todoId: number,
+    args: UpdateTodoArgs
+  ) => void;
+  updatingTodosId: number[]
+  // selectedTodoId: number | null;
+  // setSelectedTodoId: (todoId: number | null) => void;
+  updateTodoTitle: (todoId: number, args: UpdateTodoArgs) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  toggleTodoStatus,
+  updatingTodosId,
+  // selectedTodoId,
+  // setSelectedTodoId,
+  updateTodoTitle,
+
+}) => {
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+  const [isDoubleClicked, setIsDoubleClicked] = useState(false);
+
+  const inputField = useRef<HTMLInputElement>(null);
+  const isUpdatingItem = updatingTodosId.includes(todo.id);
+
+  useEffect(() => {
+    if (inputField.current) {
+      inputField.current.focus();
+    }
+  }, [isDoubleClicked]);
+
+  const clearSelectedTodoId = () => {
+    setIsDoubleClicked(false);
+  };
+
+  const resetTitle = () => setEditedTitle(todo.title);
+
+  const handleToggleTodoStatus = () => toggleTodoStatus(
+    todo.id,
+    { completed: !todo.completed },
+  );
+
+  const handleDoubleClick = (
+    event: React.MouseEvent<HTMLSpanElement>,
+  ) => {
+    event.preventDefault();
+    setIsDoubleClicked(true);
+  };
+
+  const handleInputChanges = (event:React.ChangeEvent<HTMLInputElement>) => (
+    setEditedTitle(event.target.value)
+  );
+
+  const changeTittleIfEdited = () => {
+    if (editedTitle !== todo.title) {
+      updateTodoTitle(todo.id, { title: editedTitle });
+    }
+  };
+
+  const handleOnBlur = () => {
+    changeTittleIfEdited();
+
+    clearSelectedTodoId();
+  };
+
+  const handleSubmit = (event:React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    changeTittleIfEdited();
+
+    clearSelectedTodoId();
+  };
+
+  const handleCancelEditing = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      clearSelectedTodoId();
+      resetTitle();
+    }
```

Comment:
```suggestion
    if (event.key !== 'Escape') {
      return;
    }
    
    clearSelectedTodoId();
    resetTitle();
```

in case when you handle only one button it will be clearer, because if you increase amount of logic you'll have less nesting